crates/vim/src/normal/change.rs

```diff
@@ -53,7 +54,22 @@ pub fn change_motion(vim: &mut Vim, motion: Motion, times: Option<usize>, cx: &m
                 });
             });
             copy_selections_content(vim, editor, motion.linewise(), cx);
-            editor.insert("", cx);
+
+            let (display_map, selections) = editor.selections.all_adjusted_display(cx);
+            let mut edits = Vec::new();
+            for selection in &selections {
+                let mut insert = "".to_string();
+                let offset_range = selection
+                    .map(|p| p.to_offset(&display_map, Bias::Left))
+                    .range();
+
+                if motion == Motion::CurrentLine {
+                    let (indent, _) = display_map.line_indent(selection.start.row());
+                    insert.push_str(&" ".repeat(indent as usize));
```

Comment:
I don't think we can do that, since the file might be use tabs for indentation.

Here's Neovim with a Go file, for example:

https://github.com/zed-industries/zed/assets/1185253/ee2b72af-e080-45ea-9ddd-0ca95e8c7f83


I wonder whether we shouldn't turn this around: instead of deleting and then inserting something, why not first find the first non-space character of the line and change from there to the end?

Reply:
Thanks for the reminder! I came up with an easy way to do it with a shorter code